### PR TITLE
Fix next_unique

### PR DESCRIPTION
**Pull request by okdshin**
When `initializer_list` has numerical name, `next_unique` conflicts with it. The conflict causes breaking graph.
This PR fix the part depending on `next_unique`. See also #2613.

### CODE REVIEW

File: onnx/common/ir.h [hunk 1]
```diff
@@ -866,6 +866,29 @@ friend struct Value;
 
   std::vector <OpSetID> opset_versions_;
 
+  bool isNameInAnyNode(const std::string& name) const {
+      auto f = [&name](const Value* v){ return v->uniqueName() == name; };
+      for(const Node* node : all_nodes) {
+          auto found_in = std::find_if(node->inputs().begin(), node->inputs().end(), f);
+          auto found_out = std::find_if(node->outputs().begin(), node->outputs().end(), f);
+          if(found_in != node->inputs().end() || found_out != node->outputs().end()) {
```
Comment (askhade) [inline onnx/common/ir.h:874]: nit: If you do the check for found_in before checking the node outputs then you can exit early when the name is found in node inputs.

**Comment by askhade** [comment]
Overall looks good to me. There is a small comment I added apart from that once you sign the DCO we can check this in. Thanks!

File: onnx/common/ir.h [hunk 2]
```diff
@@ -866,6 +866,29 @@ friend struct Value;
 
   std::vector <OpSetID> opset_versions_;
 
+  bool isNameInAnyNode(const std::string& name) const {
```
Comment (gramalingam) [inline onnx/common/ir.h:869]: What happens when we have nested-scopes produced via control-flow operators? Don't you want to ensure uniqueness handles that too?

**Comment by daquexian** [comment]
@askhade @gramalingam Since it is important to onnx optimizer, I updated it myself in https://github.com/onnx/onnx/pull/3071

**Comment by daquexian** [comment]
The content of this PR has been merged as #3071.

**Pull request closed without merging**